keps/sig-scheduling/20200114-taint-based-evictions.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Promote Taint Based Evictions to GA
+authors:
+  - "@damemi"
+owning-sig: sig-scheduling
+participating-sigs:
+  - sig-scheduling
+reviewers:
+  - "@Huang-Wei"
+approvers:
+  - "@liggitt"
+editor:
+creation-date: 2020-01-14
+last-updated: 2020-01-14
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Promote Taint Based Evictions to GA
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Graduation Criteria](#graduation-criteria)
+- [Testing Plan](#testing-plan)
+  - [Unit tests](#unit-tests)
+  - [Integration tests](#integration-tests)
+  - [e2e tests](#e2e-tests)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+Taint Based Evictions was introduced as an alpha feature in Kubernetes 1.6 and was promoted to
+beta in 1.13. Taint Based Evictions evicts pods from a node based on taints applied to the node
```

Comment:
It's the tricky part...

Long time ago, when the feature was proposed, the feature was supposed to evict pods based on node conditions. But along with the development, it's finalized with a much smaller scope:

> Apply `not-ready` and `unreachable` taints to the nodes, with a `NoExecute` effect. While TaintNodeByCondition feature apply taints with a `NoSchedule` effect. Sometime people confuse with them.

Other functionalities such as evicting pods and setting tolerationSeconds are out of TaintBasedEviction's scope (by taint manager and default toleration seconds admission controller).

So the name is very confusing, it's worth highlighting the tricky parts.

Reply:
If we disable TaintBasedEviction, the eviction behavior and setting default toleration seconds still work. Which also prove that those are not the scope of TaintBasedEviction.